src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -99,14 +101,20 @@ def _build_tool_pex(self, tool_subsystem, interpreter, pex_path):
 
   def execute(self):
     tool_subsystem = self.tool_subsystem_cls.scoped_instance(self)
-    pex_name = tool_subsystem.options_scope
-    pex_path = os.path.join(self.workdir, self.fingerprint, '{}.pex'.format(pex_name))
+    specs_fingerprint = stable_json_sha1(tool_subsystem.get_requirement_specs())
```

Comment:
I wasn't sure how to effectively fingerprint this and should have made a note of that. Under the assumption that the requirements are going to be static enough to not blow up the user's disk space with the pants cache, it might be reasonable to assume the ordering will remain static as well?

Reply:
Yeah, I think its reasonable to risk too-fine-grained-caching. In practice the lists are static as you suggest, and when mutated almost certainly to change deps and not just permute their order.